app/exerlog/lib/src/feature/authentication/view/landing_screen.dart

```diff
@@ -0,0 +1,179 @@
+    super.initState();
+    _tabController = TabController(length: tabs.length, vsync: this);
+    _tabController.addListener(() {
+      _tabSwitchListener(_tabController.index);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final _controller = ref.read(AuthenticationController.controller);
+
+    ref.listen(AuthenticationController.provider, (_, state) {
+      if (state is SuccessState) {
+        Navigator.of(context).pushReplacement(
+          MaterialPageRoute(
+            builder: (context) => WorkoutPage(null),
+          ),
+        );
+      } else if (state is ErrorState) {
+        ScaffoldMessenger.of(context).showSnackBar(
+          SnackBar(
+            content: Text("Something went wrong"),
+          ),
+        );
+      }
+    });
+
+    return Scaffold(
+      backgroundColor: backgroundColor,
+      body: _BuildElevatedContainer(
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.start,
+          children: [
+            Container(
+              height: 40,
+              child: TabBar(
+                tabs: tabs,
+                controller: _tabController,
+              ),
+            ),
+            Expanded(
+              child: TabBarView(
+                controller: _tabController,
+                children: [
+                  Form(key: _loginFormKey, child: LoginForm(_controller)),
+                  Form(key: _signUpFormKey, child: SignupForm(_controller)),
+                ],
+              ),
+            ),
+
+            /// Google Sign In button
+            if (_tabIndex == 0) ...[
+              GoogleSignInButton(
+                onPressed: () async {
+                  await _controller.signInWithGoogle();
+                },
+              ),
+            ],
+            SizedBox(height: 20),
+
+            /// Login / Sign up button
+            RaisedGradientButton(
+              child: Text(
+                _tabIndex > 0 ? "Sign up" : "Login",
+                style: buttonText,
+              ),
+              width: context.width * .65,
+              onPressed: () async {
+                if (_tabIndex == 0) {
+                  if (_loginFormKey.currentState!.validate()) {
+                    await _controller.signIn();
+                  }
+                } else {
+                  if (_signUpFormKey.currentState!.validate()) {
+                    if (_controller.isSamePassword()) {
+                      await _controller.signUp();
+                    } else {
+                      ScaffoldMessenger.of(context).showSnackBar(
+                        SnackBar(
+                          content: Text("Password should match"),
+                        ),
+                      );
+                    }
+                  }
+                }
+              },
+            ),
+            SizedBox(height: 20),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class _BuildElevatedContainer extends StatelessWidget {

```

Comment:
Instead of putting it on the landing screen page. You could now put it in the local widget folder.